src/System.Globalization/tests/CultureInfo/CultureInfoCurrentCulture.cs

```diff
@@ -57,6 +57,7 @@ public void CurrentUICulture()
         }
 
         [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.Uap, "Thread cultures is not honored in UWP.")]
```

Comment:
I think we should be consistent. In other cases we are also throwing PNSE. About your concerns, I think somebody is building a static analyzer which will emit errors/warnings when using such APIs.